extensions-core/lookups-cached-global/src/main/java/org/apache/druid/server/lookup/namespace/cache/CacheScheduler.java

```diff
@@ -421,9 +443,16 @@ public String getVersion()
     @Override
     public void close()
     {
-      cacheHandler.close();
-      // Log statement after cacheHandler.close(), because logging may fail (e. g. in shutdown hooks)
-      log.debug("Closed version [%s] of %s", version, entryId);
+      // ideally, close shouldn't be called by users of VersionedCache in case of a incremental load,
+      // we do not close cacheHandler for such case.
+      if (this.incremental) {
+        log.debug("VersionedCache is of type incremental load, thus not closing cacheHandler " +
```

Comment:
I am trying to understand this, even if we `VersionedCacheBuilder` we need some kind of `if` condition on when we actually close the `CacheHandler` and when we don't.

I think that part is still not clear.

Previously, I was doing that by not calling close method in `scheduleAndWait` when IncrementalLoad was happening. Feels like that's the point in time when we need stop closing the `CacheHandler` going forward.

Reply:
@leventov sorry for later reply I spent some time on it this week. If `VersionedCache` doesn't have `close()` method at all, are you saying `tryUpdateCache()` would stop closing VersionedCache a bunch of times like it does now?.
Because in the first comment in this thread you said `VersionedCache.close()` should be a private method, in which case you would need a `if` condition to handle `doIncrementalLoad` differently in `tryUpdateCache()` or in that private `close()`.

Reply:
Not stop closing. The idea of this thread is to change visibility of close() outside `CacheScheduler`, but internal logic is essentially unchanged.

An `if` would not be needed in new `VersionedCacheBuilder`, but may still be needed in `VersionedCache`. I don't see now why I wrote that "VersionedCache doesn't have close() method at all", perhaps I confused something. (Or am confusing something now.)

Reply:
makes sense. I updated the review.